src/server/pachyderm_test.go

```diff
@@ -7883,6 +7883,211 @@ func TestStatsDeleteAll(t *testing.T) {
+	}
+	require.EqualOneOf(t, validDumps, buf.String())
+
+	// Test target-file-datums flag
+	commit, err = c.StartCommit(dataRepo, "beta")
+	require.NoError(t, err)
+	w, err = c.PutFileSplitWriter(dataRepo, "beta", "data", pfs.Delimiter_SQL, 2, 0, false)
+	require.NoError(t, err)
+	_, err = w.Write([]byte(fullPGDump))
+	require.NoError(t, err)
+	require.NoError(t, w.Close())
+	require.NoError(t, c.FinishCommit(dataRepo, commit.ID))
+
+	fileInfos, err = c.ListFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	require.Equal(t, 2, len(fileInfos))
+
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, fileInfos[0].File.Path, 0, 0, &buf))
+	expected := pgDumpHeader + strings.Join(rows[0:2], "") + pgDumpFooter
+	require.Equal(t, expected, buf.String())
+	require.Equal(t, len(expected), int(fileInfos[0].SizeBytes))
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, fileInfos[1].File.Path, 0, 0, &buf))
+	expected = pgDumpHeader + rows[2] + pgDumpFooter
+	require.Equal(t, expected, buf.String())
+	require.Equal(t, len(expected), int(fileInfos[1].SizeBytes))
+
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, "data", 0, 0, &buf))
+	require.Equal(t, pgDumpHeader+pgDumpFooter, buf.String())
+	dirFileInfo, err = c.InspectFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	require.Equal(t, dirExpectedLen, int(dirFileInfo.SizeBytes))
+
+	// Test target-file-bytes flag, which should apply to the size of just the content, not the header/footer
+	commit, err = c.StartCommit(dataRepo, "gamma")
+	require.NoError(t, err)
+	// This byte threshold should yield the same results as --target-file-datums=2 :
+	w, err = c.PutFileSplitWriter(dataRepo, "gamma", "data", pfs.Delimiter_SQL, 0, 80, false)
+	require.NoError(t, err)
+	_, err = w.Write([]byte(fullPGDump))
+	require.NoError(t, err)
+	require.NoError(t, w.Close())
+	require.NoError(t, c.FinishCommit(dataRepo, commit.ID))
+
+	fileInfos, err = c.ListFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	require.Equal(t, 2, len(fileInfos))
+
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, fileInfos[0].File.Path, 0, 0, &buf))
+	expected = pgDumpHeader + strings.Join(rows[0:2], "") + pgDumpFooter
+	require.Equal(t, expected, buf.String())
+	require.Equal(t, len(expected), int(fileInfos[0].SizeBytes))
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, fileInfos[1].File.Path, 0, 0, &buf))
+	expected = pgDumpHeader + rows[2] + pgDumpFooter
+	require.Equal(t, expected, buf.String())
+	require.Equal(t, len(expected), int(fileInfos[1].SizeBytes))
+	buf.Reset()
+	require.NoError(t, c.GetFile(commit.Repo.Name, commit.ID, "data", 0, 0, &buf))
+	require.Equal(t, pgDumpHeader+pgDumpFooter, buf.String())
+	dirFileInfo, err = c.InspectFile(commit.Repo.Name, commit.ID, "data")
+	require.NoError(t, err)
+	require.Equal(t, dirExpectedLen, int(dirFileInfo.SizeBytes))
+}
```

Comment:
Would also be good to have tests that do multiple `PutFile` to the same SQL split to append stuff to it, including doing so with a different header and after files from the previous `PutFile` have been deleted.

Reply:
It would also be good to have a test which creates multiple tables in the same repo `/table1` and `/table2` and then tries to get them out with `/**`, that should give you:

```
<table1 header>
<table1 row1>
<table1 row2>
...
<table1 footer>
<table2 header>
<table2 row1>
<table2 row2>
...
<table2 footer>
```

I think this would reveal some bugs in `GetFile`

Reply:
Indeed it did. This is now a test case (not sql, but the generic header/footer content case).